depot/app/controllers/users_controller.rb

```diff
@@ -1,8 +1,7 @@
 class UsersController < ApplicationController
   include UserLoggedIn
   before_action :set_user, only: [:show, :edit, :update, :destroy]
-  before_action :ensure_user_not_destroying_itself, only: [:destroy]
-
+  before_action :ensure_user_not_destroying_itself, only: :destroy
```

Comment:
this logic should be in model.

Reply:
this function is checking whether user is not deleting a user by which he/she is logged in to application.